piquasso/_backends/fock/pure/calculations.py

```diff
@@ -303,3 +306,71 @@ def _add_occupation_number_basis(  # type: ignore
     index = state._space.index(occupation_numbers)
 
     state._state_vector[index] = coefficient
+
+
+def measure_homodyne(
+    state: PureFockState, instruction: Instruction, shots: int
+) -> Result:
+    modes = instruction.modes
+    phi = instruction._all_params["phi"]
+    reduced_state = state.reduced(modes=modes)
+    scaled_hbar = 1 / reduced_state._config.hbar
+
+    if phi != 0:
+        phi_scaled = np.diag(np.exp(1j * np.atleast_1d(phi)))
+        fock_operator = reduced_state._space.get_passive_fock_operator(
+            phi_scaled,
+            modes=instruction.modes,
+            d=reduced_state._space.d,
+            permanent_function=reduced_state._config.permanent_function,
+        )
+        reduced_state._density_matrix = (
+            fock_operator
+            @ reduced_state._density_matrix
+            @ fock_operator.conjugate().transpose()
+        )
+    resolution = 100000
```

Comment:
This is not a good approach. You should try to do this without approximation, since there is an exact distribution one can calculate.

Also, one can compare in a test the resulting distribution with e.g. the distribution generated by the GaussianSimulator.